### PR TITLE
fix(mode): Fix bug where zoom did not work if mode was a function

### PR DESCRIPTION
**Pull request by SomeKittens**

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -76,12 +76,15 @@ function storeOriginalOptions(chart) {
...
+ * @param {Chart} chart instance of the chart in question
  */
-function directionEnabled(mode, dir) {
+function directionEnabled(mode, dir, chart) {
 	if (mode === undefined) {
 		return true;
 	} else if (typeof mode === 'string') {
 		return mode.indexOf(dir) !== -1;
+	} else if (typeof mode === 'function') {
+		return mode({ chart }).indexOf(dir) !== -1;
```
Comment (benmccann) [inline src/plugin.js]: Travis is failing. You'll have to change this to `{chart}`

**Comment by SomeKittens** [comment]
Pushed a lint fix, and added tests for this case.  Confirmed that new tests fail on master without the fix in this PR.

**Comment by SomeKittens** [comment]
Crosslinking: https://github.com/chartjs/chartjs-plugin-zoom/issues/319

**Comment by SomeKittens** [comment]
Awesome, thanks for the quick merge!  Can you publish it to npm?

**Pull request merged**